tests/eas/load_tracking.py

```diff
@@ -1155,12 +1152,6 @@ def _getExperimentsConf(cls, test_env):
     def setUpClass(cls, *args, **kwargs):
         super(TwoGroupsCascade, cls).runExperiments(*args, **kwargs)
 
-    def test_util_root_group(self):
-        """
-        Test utilization propagation to cgroup root
-        """
-        return self._test_group_util('/')
```

Comment:
Since the root group can be kind-of different then child groups... I was wondering if we can easily go for a slightly different fix where, in `_test_group_util` we use `assertGreaterEqual` instead of `assertAlmostEqual` at L1075 above.

Actually, since in `_test_group_util` there are two different checks, one for childs and one for the group itself, maybe we can have a more generic solution by passing in the assert functions are parameters, e.g.

```python
def _test_group_util(self, group, group_assert=self.assertAlmostEqual, child_assert=self.assertAlmostEqual):
```

And the test the root groups with:
```python
return self._test_group_util('/', group_assert=self.assertGreaterEqual)
```

Reply:
@Elieva  and I tested with assertGreaterEqual, and we concluded that changing the assert based on the group was just more code bloat without any real added value (unless the root group is actually  special so we want to test it just because it is the root).
Having different asserts also requires recoding a one-sided assertAlmostEqual since assertGreaterEqual fails sometimes when it is slightly smaller (which is expected).

Reply:
Mmm... have you tryied by replacing both the `assertAlmostEqual` into `_test_group_util`?

Because I think it's needed only for the top level group... and again, with the above solution we can actually use it only when we test the root group.

Reply:
What we need is to check that X is within these ranges:
* for the root group: [X - delta1; X + delta2], with delta2 > delta1
* for the other groups: [X - delta1; X + delta1]

delta2 needs to be large enough to account for tasks "randomly" executing, so it needs to be significantly larger than delta1. delta1 is still needed as the average we compute is can sometimes be slightly smaller than the measured value. assertAlmostEqual cannot be used for root group, neither can assertGreaterEqual unless you relax the test even more. All in all, I don't really see the value of writing complicated code if the root group does not have any special property. We can just test other groups with similar set of children and we will get the same coverage of utilization computation and propagation to parents.

Reply:
In your example above, why can't use assertGreaterEqual(X - delta1) for the group we call `_test_group_util()` on?

Reply:
assertGreaterEqual(X - delta1) check that [X - delta1; +infinite] instead of that: [X - delta1; X + delta2]
That is going to only catch half of the issues, that would also be caught when testing the other groups, at the cost of extra bloat to support it.

Reply:
Outcome: we settled on the fact that the root group is somehow special in the kernel (different code paths), so testing non-root groups does not provide full coverage. As a result, the root group will be checked with a "greater or equal" test to avoid issues when unexpected tasks are executing in it.